### PR TITLE
prevent underflow in output_comment_multi_simple 

### PR DESCRIPTION
**Pull request by CDanU**
ref: #1838

### CODE REVIEW

File: src/output.cpp [hunk 1]
```diff
@@ -2011,99 +2012,106 @@ static void do_kw_subst(chunk_t *pc)
...
-   else
-   {
-      // The comment starts after something else
-      col_diff = 0;
-   }
+   // The multiline comment is saved inside one chunk. If the comment is
+   // shifted all lines of the comment need to be shifter by the same amount.
+   // Save the difference of initial and current position to apply it on every
+   // line_column
+   const int col_diff = [pc](){
```
Comment (mihaipopescu) [inline src/output.cpp]: Is the lambda really required here?

File: src/output.cpp [hunk 2]
```diff
@@ -2011,99 +2012,106 @@ static void do_kw_subst(chunk_t *pc)
...
          else if (ch == '\t')
          {
-            ccol = calc_next_tab_column(ccol, cpd.settings[UO_input_tab_size].u);
+            line_column = calc_next_tab_column(line_column, cpd.settings[UO_input_tab_size].u);
             continue;
          }
-         else
+      }
+
+      // 2: add chars to line, handle the CRLF and CR endings (convert both to LF)
```
Comment (mihaipopescu) [inline src/output.cpp]: > convert both to LF

why? isn't that controlled by an option?

Comment (CDanU) [reply src/output.cpp]: I just refactored the function to make it more understandable (mostly for my own sake).  While doing that I didn't gave too much thought into actually changing its behaviour outside of what I needed to change to fix the reported issue. 

Here the `\r\n` are converted into `\n`, this most certainly isn't needed here. 
The actual determination which line ends are going to be used in the end are happening inside the end of the tokenization function (using the `UO_newlines` option) and the write char function which actually is writing the correct line ends.

File: src/output.cpp (hunk 1 above)
Comment (CDanU) [reply src/output.cpp]: Well not really but I wanted the `col_diff` to be const.

File: src/output.cpp (hunk 2 above)
Comment (CDanU) [reply src/output.cpp]: Dropping the check will force me to make nasty checks in other places. 
Why do we keep the \r around anyways? I will try to remove them in the tokenization and see if that causes a total mess or simplifies things even more.

**Comment by CDanU** [comment]
I am not entirely sure why coveralls doesn't show that all new lines are covered ... I stepped locally through the code, I do hit this block and it works as it should. The only reason that I can think of is that git somehow eats the '\r' in the push process (or fetch for the travis run) but that shouldn't happen.

**Pull request merged**